### PR TITLE
Update ember-testing tests to new syntax

### PR DESCRIPTION
**Pull request by cibernox**
Part of #15988

### CODE REVIEW

File: packages/ember-testing/tests/adapters/qunit_test.js [hunk 1]
```diff
@@ -1,51 +1,55 @@
...
-    QUnit.stop = function() {
-      ok(true, 'stop called');
-    };
-    adapter.asyncStart();
-  } finally {
-    QUnit.stop = originalStop;
+  ['@test asyncStart calls stop'](assert) {
+    var originalStop = QUnit.stop;
+    try {
+      QUnit.stop = function() {
```
Comment (cibernox) [inline packages/ember-testing/tests/adapters/qunit_test.js]: I'm not very sure what to do about `QUnit.stop` and `QUnit.start`

File: packages/ember-testing/tests/adapters/qunit_test.js [hunk 2]
```diff
@@ -1,51 +1,55 @@
...
-  var originalStart = QUnit.start;
-  try {
-    QUnit.start = function() {
-      ok(true, 'start called');
-    };
-    adapter.asyncEnd();
-  } finally {
-    QUnit.start = originalStart;
+  ['@test asyncEnd calls start'](assert) {
+    var originalStart = QUnit.start;
```
Comment (cibernox) [inline packages/ember-testing/tests/adapters/qunit_test.js]: Again

File: packages/ember-testing/tests/adapters/qunit_test.js [hunk 3]
```diff
@@ -1,51 +1,55 @@
...
-      originalOk(msg, '{err: "hai"}');
-    };
-    adapter.exception(error);
-  } finally {
-    window.ok = originalOk;
+  ['@test exception causes a failing assertion']() {
+    var error = { err: 'hai' };
+    var originalOk = window.ok;
+    try {
+      window.ok = function(val, msg) {
```
Comment (cibernox) [inline packages/ember-testing/tests/adapters/qunit_test.js]: This is also unclear.

File: packages/ember-testing/tests/ext/rsvp_test.js [hunk 4]
```diff
@@ -23,97 +25,101 @@ QUnit.module('ember-testing RSVP', {
...
+        assert.equal(asyncEnded, 1);
+      });
+    }).then(function() {
+      assert.equal(asyncStarted, 1);
+      assert.equal(asyncEnded, 1);
+
+      return new RSVP.Promise(function(resolve) {
+        QUnit.stop(); // raw async, we must inform the test framework manually
+        setTimeout(function() {
+          QUnit.start(); // raw async, we must inform the test framework manually
```
Comment (cibernox) [inline packages/ember-testing/tests/ext/rsvp_test.js]: again

File: packages/ember-testing/tests/adapters/qunit_test.js (hunk 1 above)
Comment (rwjblue) [reply packages/ember-testing/tests/adapters/qunit_test.js]: The places that are explicitly testing `asyncStart` and `asyncStop` should continue to use `QUnit.stop` / `QUnit.start` for now, but anywhere else it is being done should be migrated to `assert.async()`.

File: packages/ember-testing/tests/adapters/qunit_test.js (hunk 3 above)
Comment (rwjblue) [reply packages/ember-testing/tests/adapters/qunit_test.js]: I have some tests for this in ember-qunit, basically you should do:

```js
    let originalPushResult = assert.pushResult;
    assert.pushResult = function(resultInfo) {
      // Inverts the result so we can test failing assertions
      resultInfo.result = !resultInfo.result;
      resultInfo.message = `Failed: ${resultInfo.message}`;
      originalPushResult(resultInfo);
    };
```

File: packages/ember-testing/tests/ext/rsvp_test.js (hunk 4 above)
Comment (rwjblue) [reply packages/ember-testing/tests/ext/rsvp_test.js]: Should use `assert.async();` here

**Comment by rwjblue** [comment]
@cibernox - Had any time to work on this?

**Comment by cibernox** [comment]
I do now. TBH, i just forgot about this PR 😅

File: packages/ember-testing/lib/adapters/qunit.js [hunk 5]
```diff
@@ -18,6 +18,6 @@ export default Adapter.extend({
     QUnit.start();
   },
   exception(error) {
-    ok(false, inspect(error));
+    ok(false, inspect(error)); // How do we access assert.ok here?
```
Comment (cibernox) [inline packages/ember-testing/lib/adapters/qunit.js]: @rwjblue Should I use `QUnit.assert.ok` here instead?

Comment (rwjblue) [reply packages/ember-testing/lib/adapters/qunit.js]: `QUnit.config.current.assert.ok`

**Comment by cibernox** [comment]
@rwjblue ready

**Review by rwjblue — APPROVED** [review]

**Pull request merged**